### PR TITLE
dont show tx history older than 6 months

### PR DESCRIPTION
**Pull request by Kevingislason**
## Description

Don't show transaction history older than 6 months; this was a missed requirement for the original transaction history feature.

---

Before we can merge this PR, please make sure that all the following items have been checked off. If any of the checklist items are not applicable, please leave them but write a little note why.

- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work
- [x] Re-reviewed files changed in the Github PR explorer

## If PR to `staging`
- [x] Updated relevant documentation in the code
- [x] Test code changes manually, and describe your procedure in a comment on this PR
- [x] Mark this PR with the correct milestone

### CODE REVIEW

**Comment by jincan39** [comment]
LGTM

**Review by jincan39 — APPROVED** [review]

File: src/utils/persistence/privateTransactionHistory.ts [hunk 1]
```diff
@@ -7,11 +7,17 @@ export const getPrivateTransactionHistory = (): TxHistoryEvent[] => {
...
+  .map(
     (jsonTxHistoryEvent): TxHistoryEvent => {
       return TxHistoryEvent.fromJson(jsonTxHistoryEvent);
     }
-  );
+  )
+  .filter((event): TxHistoryEvent => {
+    const sixMonthsAgo = new Date();
+    sixMonthsAgo.setMonth(sixMonthsAgo.getMonth() - 6);
+    return sixMonthsAgo < event.date;
```
Comment (jincan39) [inline src/utils/persistence/privateTransactionHistory.ts:19]: sixMonthsAgo 's type not match with TxHistoryEvent.date's type
`return sixMonthsAgo < event.date.valueOf();`

**Review by jincan39 — CHANGES_REQUESTED** [review]

File: src/utils/persistence/privateTransactionHistory.ts [hunk 2]
```diff
@@ -7,11 +7,17 @@ export const getPrivateTransactionHistory = (): TxHistoryEvent[] => {
...
   ];
-  const privateTransactionHistory = jsonPrivateTransactionHistory.map(
+  const privateTransactionHistory = jsonPrivateTransactionHistory
+  .map(
     (jsonTxHistoryEvent): TxHistoryEvent => {
       return TxHistoryEvent.fromJson(jsonTxHistoryEvent);
     }
-  );
+  )
+  .filter((event): TxHistoryEvent => {
```
Comment (jincan39) [inline src/utils/persistence/privateTransactionHistory.ts:16]: return type not matched
think it should be like `.filter((event: TxHistoryEvent): boolean => `

**Review by jincan39 — CHANGES_REQUESTED** [review]

File: src/utils/persistence/privateTransactionHistory.ts (hunk 2 above)
Comment (zhangjing28) [reply src/utils/persistence/privateTransactionHistory.ts:16]: Agree with Jincan here, actually, you don't have to add the return type of the `map` and `filter` callback function. Typescript is smart enough to know it

**Review by zhangjing28 — APPROVED** [review]

**Review by jincan39 — APPROVED** [review]

**Review by jasonintju — APPROVED** [review]

**Pull request merged**